### PR TITLE
Multi-subnet clusters: only add nodes from the default network to the shared instance group

### PR DESCRIPTION
**Pull request by mmamczur**

### CODE REVIEW

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -512,6 +530,26 @@ var (
 	}
 )
 
+func CandidateNodesInDefaultSubnetPredicate(defaultSubnet string) func(node *api_v1.Node, logger klog.Logger) bool {
+	return func(node *api_v1.Node, logger klog.Logger) bool {
+		predicate := nodePredicateInternal(node, false, false, logger)
```
Comment (sawsa307) [inline pkg/utils/utils.go:535]: can we do `if !CandidateNodesPredicate(node, logger)` here?

Comment (mmamczur) [reply pkg/utils/utils.go]: changed

File: pkg/utils/utils.go [hunk 2]
```diff
@@ -512,6 +530,25 @@ var (
...
+		}
+		gkeSubnetworkLabel, hasGKESubnetworkLabel := node.Labels[LabelGKESubnetworkName]
+		// if there is no label but the podCIDR is populated assume default network
+		if !hasGKESubnetworkLabel && node.Spec.PodCIDR != "" {
+			return true
+		}
+		if hasGKESubnetworkLabel && EqualResourceIDs(gkeSubnetworkLabel, defaultSubnet) {
+			return true
+		}
+		logger.Info("Skipping node for the instance group since it is from a non-default network", "node", node.Name, "nodeSubnetwork", gkeSubnetworkLabel)
```
Comment (sawsa307) [inline pkg/utils/utils.go:546]: Wondering if we can have separate logs for `Node without PodCIDR` and `Node from non-default subnet`? Right now they would both emit this same log line and can be confusing during debugging

Comment (sawsa307) [reply pkg/utils/utils.go:546]: I just created a PR #2534 to generalize the way we do node checking. If we use the same helper functions, it is also easier to grep for logs.
 Let me know if the PR makes sense to you, thanks!

File: pkg/utils/utils.go [hunk 3]
```diff
@@ -481,6 +484,21 @@ func GetReadyNodeNames(lister listers.NodeLister, logger klog.Logger) ([]string,
 	return nodeNames, nil
 }
 
+// GetReadyNodeNamesInDefaultSubnet returns names of schedulable, ready nodes from the node lister
```
Comment (cezarygerard) [inline pkg/utils/utils.go:487]: is it possible to move
GetReadyNodeNamesInDefaultSubnet
and
GetReadyNodeNames

to pkg/instancegroups/nodegetter.go

maybe in separate PR

**Comment by cezarygerard** [comment]
lgtm overall

**Pull request closed without merging**